apps/api/src/app/change/usecases/change-enabled-notification-template/change-enabled-notification-template.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { TypeChangeEnabledCommand } from '../type-change-enabled.command';
+import { NotificationTemplateEntity, NotificationTemplateRepository, MessageTemplateRepository } from '@novu/dal';
+
+@Injectable()
+export class ChangeEnabledNotificationTemplate {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private messageTemplateRepository: MessageTemplateRepository
+  ) {}
+
+  async execute(command: TypeChangeEnabledCommand) {
+    const item = await this.notificationTemplateRepository.findOne({
+      _environmentId: command.environmentId,
+      _parentId: command.item._id,
+    });
+
+    const newItem = command.item as NotificationTemplateEntity;
+
+    const messages = await this.messageTemplateRepository.find({
+      _environmentId: command.environmentId,
+      _parentId: {
+        $in: newItem.steps ? newItem.steps.map((step) => step._id) : [],
+      },
+    });
+
+    const missingMessages = [];
+    const steps = newItem.steps
+      ? newItem.steps
+          .map((step) => {
+            const oldMessage = messages.reduce((prev, message) => {
+              return message._parentId === step._id ? message : prev;
+            }, undefined);
```

Comment:
Not sure I clearly understand the need of reduce here, should it just find the parentId? 
```suggestion
            const oldMessage = messages.find((message) => {
              return message._parentId === step._id;
            });
```

Or does it something more than that?